### DISCUSSION TITLE
As a module -  Alpine.start() from DOMContentLoaded?

### DISCUSSION THREAD

**Discussion by mfazekas** [alpinejs/alpine#2274] [1. Help] [answered]
The docs at
https://alpinejs.dev/essentials/installation#as-a-module

says use the following when loading alpine from a bundle.

```js
import Alpine from 'alpinejs'
 
window.Alpine = Alpine
 
Alpine.start()
```

the above in our rails/webpacker app gives the following warning:

```js
Alpine Warning: Unable to initialize. Trying to load Alpine before `<body>` is available. Did you forget to add `defer` in Alpine's `<script>` tag?
```

Moving Alpine.start() into 

```js
document.addEventListener("DOMContentLoaded", function(event) {
  Alpine.start();
});
```

fixes this, but is it the right place to execute `Apline.start()`? Would interactive readyState be any better?

**Comment by SimoTod** [comment, marked as answer]
Generally speaking Alpine needs to be deferred, or loaded at the bottom of your body. If it's not possible, you can use any strategy suits you, it just needs to start after the whole DOM is available (so yeah, DOMcontentLoaded is a valid option).
